rotkehlchen/exchanges/binance.py

```diff
@@ -522,30 +529,91 @@ def _deserialize_asset_movement(self, raw_data: Dict[str, Any]) -> Optional[Asse
 
         return None
 
+    def _api_query_dict_within_time_delta(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            time_delta: Timestamp,
+            method: Literal["depositHistory.html", "withdrawHistory.html"],
+    ) -> List[Dict[str, Any]]:
+        """Request via `api_query_dict()` from `start_ts` `end_ts` using a time
+        delta (offset) less than `time_delta`.
+
+        Be aware of:
+          - If `start_ts` equals zero, the Binance launch timestamp is used
+          (from BINANCE_LAUNCH_TS). This value is not stored in the
+          `used_query_ranges` table, but 0.
+          - Timestamps are converted to milliseconds.
+        """
+        if method == 'depositHistory.html':
+            query_schema = 'depositList'
+        elif method == 'withdrawHistory.html':
+            query_schema = 'withdrawList'
+        else:
+            raise AssertionError(f'Unexpected binance method case: {method}.')
+
+        results: List[Dict[str, Any]] = []
+        # Create required time references in milliseconds
+        offset = time_delta * 1000 - 1  # less than time_delta
+        if start_ts == Timestamp(0):
+            from_ts = BINANCE_LAUNCH_TS * 1000
+        else:
+            from_ts = start_ts * 1000
+
+        to_ts = (
+            from_ts + offset  # Case request with offset
+            if end_ts * 1000 - from_ts > offset
+            else end_ts * 1000  # Case request without offset (1 request)
+        )
+        while True:
+            options = {
+                'timestamp': ts_now_in_ms(),
+                'startTime': from_ts,
+                'endTime': to_ts,
+            }
+            result = self.api_query_dict(method, options=options)
+            results.extend(result.get(query_schema, []))
+            # Case stop requesting
+            if to_ts == end_ts * 1000:
```

Comment:
it does not hurt to be safe. Make this `>=`. Let's not be hit by an of by one error turning this loop into an infinite loop.

Reply:
True, good one